src/aws/flb_aws_util.c

```diff
@@ -507,3 +512,181 @@ int flb_read_file(const char *path, char **out_buf, size_t *out_size)
 
     return 0;
 }
+
+/* Generic replace function for strings. */
+char* replace_uri_tokens(const char* original_string, const char* current_word, 
```

Comment:
Small thing: Add `static` in front of all functions that aren't exported in the header file:

```
static char* replace_uri_tokens(const char* original_string, const char* current_word...
```

This makes them private.